action.yml

```
@@ -29,12 +29,35 @@ runs:
 
             const idToken = await core.getIDToken(aud)
 
-            const providerTokenResponse = await fetch(`${apiBaseURL}/tokens/auth/login`, {
-              method: "POST",
-              headers: {
-                "Authorization": `Bearer ${idToken}`
+            const isRetryableError = (response) => {
+              return response.status >= 500 || response.status === 429
+            }
+
+            const loginWithRetries = async (tries) => {
+              const providerTokenResponse = await fetch(`${apiBaseURL}/tokens/auth/login`, {
+                method: "POST",
+                headers: {
+                  "Authorization": `Bearer ${idToken}`
+                }
+              })
+
+              if (providerTokenResponse.ok) {
+                return providerTokenResponse
+              } else {
+                if (tries > 0 && isRetryableError(providerTokenResponse)) {
+                  console.log(`Failed to get provider token: ${providerTokenResponse.status} ${providerTokenResponse.statusText}. Retrying...`)
+
+                  // Random backoff between 0 and 3 seconds
+                  await new Promise(resolve => setTimeout(resolve, Math.floor(Math.random() * 3000)))
+
+                  return loginWithRetries(tries - 1)
+                } else {
+                  throw new Error(`Failed to get provider token: ${providerTokenResponse.status} ${providerTokenResponse.statusText}`)
```

We'll capture the last response code and log it as usual rather than going into the catch block
```suggestion
                  return providerTokenResponse
```